### PR TITLE
Integrate Geogebra Designer - first steps MATHNETNG-38 MATHNETNG-7

### PR DESCRIPTION
**Pull request by Polej**
First steps towards integrating Geogebra Designer inside Vue component

### CODE REVIEW

File: src/components/Geogebra/GeogebraInterface.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// import GGBApplet from '../../vendor/js/deployggb';
+
+export default class {
+    constructor(domElementId) {
+        this.params = {
+            container: domElementId,
+            id: 'applet',
```
Comment (IwoHerka) [inline src/components/Geogebra/GeogebraInterface.js:5]: This should be configurable.

File: src/components/Geogebra/GeogebraInterface.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            width: 800,
```
Comment (IwoHerka) [inline src/components/Geogebra/GeogebraInterface.js:6]: This and height too.

File: src/components/Geogebra/GeogebraInterface.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    inject(callback) {
+        window.ggbOnInit = () => {
+            this.ggbOnInit();
+            callback(); 
+        };
+
+        this.appletContainer.inject(this.params.container, 'auto');
+
+        console.log('Inject!');
```
Comment (IwoHerka) [inline src/components/Geogebra/GeogebraInterface.js]: Someone forgot about test log. :D

File: src/components/Pages/Designer.vue [hunk 4]
```diff
@@ -67,7 +67,7 @@
                             <div class="row mt-5">
                                 <div class="col-12">
                                     <button class="btn btn-warning p-2">Reset view</button>
-                                    <div class="geogebra-applet">
+                                    <div id="geogebraDesigner" class="geogebra-applet">
```
Comment (IwoHerka) [inline src/components/Pages/Designer.vue]: For DOM ids use snake_case.

File: src/components/Pages/Designer.vue [hunk 5]
```diff
@@ -197,6 +197,20 @@
...
+import GeogebraInterface from '../Geogebra/GeogebraInterface';
+
 export default {
+    mounted() {
+        // simple example code to show how to initialize GeoGebra
+        const GI = new GeogebraInterface('geogebraDesigner'); // initializing our custom class
+        GI.inject( () => { // passing callback
+            const xml = GI.getXML(); // getting Geogebra state
+            // eslint-disable-next-line no-console
+            console.log('XML is ', xml); // resulting xml
```
Comment (IwoHerka) [inline src/components/Pages/Designer.vue]: -> log

File: src/components/Pages/Designer.vue [hunk 6]
```diff
@@ -197,6 +197,20 @@
 </template>
 
 <script>
+import GeogebraInterface from '../Geogebra/GeogebraInterface';
+
 export default {
+    mounted() {
+        // simple example code to show how to initialize GeoGebra
+        const GI = new GeogebraInterface('geogebraDesigner'); // initializing our custom class
```
Comment (IwoHerka) [inline src/components/Pages/Designer.vue]: One blank life before entering new block.

File: src/components/Pages/Designer.vue [hunk 7]
```diff
@@ -197,6 +197,20 @@
...
+        GI.inject( () => { // passing callback
```
Comment (IwoHerka) [inline src/components/Pages/Designer.vue]: Whitespace before argument.

File: src/components/Pages/Designer.vue [hunk 8]
```diff
@@ -197,6 +197,20 @@
...
+            const xml = GI.getXML(); // getting Geogebra state
+            // eslint-disable-next-line no-console
+            console.log('XML is ', xml); // resulting xml
+            GI.setXML(xml); // setting Geogebra state from xml
+        });
+        // eslint-disable-next-line no-console
+        console.log(GI);
```
Comment (IwoHerka) [inline src/components/Pages/Designer.vue]: -> log

**Review by IwoHerka — CHANGES_REQUESTED** [review]

File: src/components/Pages/Designer.vue (hunk 6 above)
Comment (Polej) [reply src/components/Pages/Designer.vue]: between export default and mounted() ?

File: src/components/Pages/Designer.vue (hunk 7 above)
Comment (Polej) [reply src/components/Pages/Designer.vue]: you mean in inject( () ... ?

**Pull request merged**